src/libsyntax_ext/deriving/cmp/partial_eq.rs

```diff
@@ -27,29 +27,25 @@ pub fn expand_deriving_partial_eq(cx: &mut ExtCtxt,
     // structures are equal if all fields are equal, and non equal, if
     // any fields are not equal or if the enum variants are different
     fn cs_eq(cx: &mut ExtCtxt, span: Span, substr: &Substructure) -> P<Expr> {
-        cs_fold1(true, // use foldl
-            |cx, span, subexpr, self_f, other_fs| {
-                let other_f = match (other_fs.len(), other_fs.get(0)) {
-                    (1, Some(o_f)) => o_f,
-                    _ => cx.span_bug(span, "not exactly 2 arguments in `derive(PartialEq)`"),
-                };
+        let eq = |cx: &mut ExtCtxt, span: Span, self_f: P<Expr>, other_fs: &[P<Expr>]| {
```

Comment:
I meant splitting this out into a function such that it can be shared between the Eq and Neq code

Reply:
Ahh... that's another good idea! 😅